utils/dependency_walker.go

```diff
@@ -0,0 +1,225 @@
+package utils
+
+import (
+	"fmt"
+
+	"github.com/VirusTotal/gyp/ast"
+)
+
+var conditionKeywords = []string{"int8", "int16", "int32", "uint8", "uint16", "uint32", "int8be", "int16be", "int32be", "uint8be", "uint16be", "uint32be"}
```

Comment:
This is mainly for identifying built-in functions like `intXX` from other identifiers, so I think it's worth improving the parser by including more information about each identifier in the AST itself. The need for this array comes from a lack of features in the AST. Each `ast.Identifier` struct could contain not only the name, but also the kind of identifier if known. During the parsing we are able to know if an identifier is a rule identifier, or a built-in function, for example.